src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -490,13 +497,15 @@ void lmdb_resized(MDB_env *env)
   MGINFO("LMDB Mapsize increased." << "  Old: " << old / (1024 * 1024) << "MiB" << ", New: " << new_mapsize / (1024 * 1024) << "MiB");
 
   mdb_txn_safe::allow_new_txns();
+  if (isactive)
+    mdb_txn_safe::increment_txns(1);
```

Comment:
Perhaps it's a better idea to put this back right after `wait_no_active_txns()`. Because, if that exception is thrown, and it's caught, and then the `mdb_txn_safe` object is destroyed, it'll lead to `num_active_txns` of `-1`.

Alternatively, something like `BOOST_SCOPE_EXIT` can be used to ensure that RAII will make the value go back up before the function ends.